velox/functions/prestosql/ArrayIntersectExcept.cpp

```diff
@@ -286,6 +286,10 @@ class ArraysOverlapFunction : public exec::VectorFunction {
       auto offset = baseLeftArray->offsetAt(idx);
       auto size = baseLeftArray->sizeAt(idx);
       bool hasNull = rightSet.hasNull;
+      if (size == 0 || (rightSet.set.size() == 0 && !hasNull)) {
```

Comment:
perhaps, add a helper method 'empty()' to SetWithNull to make this code a bit easier to read

```
if (size == 0 || rightSet.empty())
```

I assume this check says "If either side is empty".

Reply:
Updated, thanks.